pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -616,26 +617,48 @@ func (jm *ControllerV2) cleanupFinishedJobs(cj *batchv1beta1.CronJob, js []batch
 	}
 
 	if cj.Spec.SuccessfulJobsHistoryLimit != nil {
-		removeOldestJobs(cj,
+		jm.removeOldestJobs(cj,
 			successfulJobs,
-			jm.jobControl,
-			*cj.Spec.SuccessfulJobsHistoryLimit,
-			jm.recorder)
+			*cj.Spec.SuccessfulJobsHistoryLimit)
 	}
 
 	if cj.Spec.FailedJobsHistoryLimit != nil {
-		removeOldestJobs(cj,
+		jm.removeOldestJobs(cj,
 			failedJobs,
-			jm.jobControl,
-			*cj.Spec.FailedJobsHistoryLimit,
-			jm.recorder)
+			*cj.Spec.FailedJobsHistoryLimit)
 	}
 
 	// Update the CronJob, in case jobs were removed from the list.
 	_, err := jm.cronJobControl.UpdateStatus(cj)
 	return err
 }
 
+func (jm *ControllerV2) getFinishedStatus(j *batchv1.Job) (bool, batchv1.JobConditionType) {
+	for _, c := range j.Status.Conditions {
+		if (c.Type == batchv1.JobComplete || c.Type == batchv1.JobFailed) && c.Status == corev1.ConditionTrue {
+			return true, c.Type
+		}
+	}
+	return false, ""
+}
+
+// removeOldestJobs removes the oldest jobs from a list of jobs
+func (jm *ControllerV2) removeOldestJobs(cj *batchv1beta1.CronJob, js []*batchv1.Job, maxJobs int32) {
```

Comment:
Yeah, that one has to be duplicated since you have different arguments, namely `[]*batchv1.Job` here vs `[]batchv1.Job` in the other controller file.